components/map/map.jsx

```diff
@@ -124,6 +152,55 @@ export default class Map extends React.Component {
         this.props.onPokemonClick(randomPokemon);
       }
     });
+
+    setTimeout(() => this.updatePokemonPosition(pokemonMarker, map), 1);
+    setInterval(() => this.updatePokemonPosition(pokemonMarker, map), 800);
+
+    map.controls[google.maps.ControlPosition.TOP_RIGHT].push(this.runawayButton(map, pokemonMarker));
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(this.pokemonMessage(map, pokemonMarker));
+  }
+
+  updatePokemonPosition(pokemonMarker, map) {
+    const latDiff = 0.18;
+    const lngDiff = 0.40;
+    const maxLat = map.getCenter().lat() + latDiff;
+    const minLat = map.getCenter().lat() - latDiff;
+    const maxLng = map.getCenter().lng() + lngDiff;
+    const minLng = map.getCenter().lng() - lngDiff;
+    const newLat = Math.random() * (maxLat - minLat) + minLat;
+    const newLng = Math.random() * (maxLng - minLng) + minLng;
+    const newPosition = new google.maps.LatLng(newLat, newLng);
+    pokemonMarker.setPosition(newPosition);
+  }
+
+  pokemonMessage(map, pokemonMarker) {
+    const pokemonMessage = $(
+      `<div id="pokemon-message">A wild ${this.state.pokemon.name} appeared! Click to catch it!</div>`
```

Comment:
It'll be better if you added the styles inside so that when it adds the DOMNode in it renders it with the styles. So <div id={} style="margin: ...">